lib/puppet/provider/user/aix.rb

```diff
@@ -265,6 +265,26 @@ def create
     end
   end
 
+  # Deletes this instance resource
+  def delete
+    homedir = home
+    super
+    if @resource.managehome?
```

Comment:
could use an early return since you don't need an else branch

```ruby
return unless @resource.managehome?

if !Puppet::Util.absolute_path?(homedir) ...
```